### PR TITLE
Liqo Agent doc

### PR DESCRIPTION
**Pull request by AbakusW**
# Liqo Agent documentation

This PR adds the documentation for the Liqo Agent component

### CODE REVIEW

File: docs/design/Liqo-Agent.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# Liqo Agent
+## Overview
+**Liqo Agent** is the Liqo component that lets the user interact with Liqo in an easy, friendly way.
+It represents one of the two main entry point to the Liqo environment, along with the **LiqoDash**.
+
+_Agent_ is accessible from its tray icon and provides a menu trough which users can:
+
+* monitor the **status** of their own Liqo
+* get **quick access** to Liqo **main features**, e.g. start/stop, Liqo working mode,
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:9]: Working mode are not presented in this document, and I feel that we need them at a certain point to make readers understand what it is about.

File: docs/design/Liqo-Agent.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# Liqo Agent
+## Overview
+**Liqo Agent** is the Liqo component that lets the user interact with Liqo in an easy, friendly way.
+It represents one of the two main entry point to the Liqo environment, along with the **LiqoDash**.
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:4]: Missing reference to LiqoDash repository

File: docs/design/Liqo-Agent.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Liqo Agent
+## Overview
+**Liqo Agent** is the Liqo component that lets the user interact with Liqo in an easy, friendly way.
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:3]: I would say that is thought in particular for desktop users.

File: docs/design/Liqo-Agent.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    * **look**, e.g. visibility, text content, possibility to be clickable
+    * **functionality**, managing the connection of the item with a callback
+
+* **Agent Controller**: It is the component that manages _Agent_ interactions with the Liqo cluster.
+It embeds all the necessary logic and data structures (like clients and caches) in order to 
+operate with the Liqo CRDs, watching relevant events and signaling them to the _Indicator_ Notify system.
+
+### Workflow
+
+* When _Agents_ starts:
```
Comment (fraborg) [inline docs/design/Liqo-Agent.md:87]: I would use a numbered list since these are sequential steps

File: docs/design/Liqo-Agent.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+It represents one of the two main entry point to the Liqo environment, along with the **LiqoDash**.
+
+_Agent_ is accessible from its tray icon and provides a menu trough which users can:
+
+* monitor the **status** of their own Liqo
+* get **quick access** to Liqo **main features**, e.g. start/stop, Liqo working mode,
+peering management
+* manage the Agent settings, e.g. notifications
+
+_Agent_ aims at enhancing the Liqo users' experience, especially for non-pro ones,
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:13]: I would replace non-pro with desktop users.

File: docs/design/Liqo-Agent.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+_Agent_ is accessible from its tray icon and provides a menu trough which users can:
+
+* monitor the **status** of their own Liqo
+* get **quick access** to Liqo **main features**, e.g. start/stop, Liqo working mode,
+peering management
+* manage the Agent settings, e.g. notifications
+
+_Agent_ aims at enhancing the Liqo users' experience, especially for non-pro ones, 
+allowing them to perform simple operations without needing to interact with the **LiqoDash**.
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:14]: I would replace LiqoDash with Kubectl. Normally, everybody can use a dashboard without training, but will require some exercise to deal with kubectl

File: docs/design/Liqo-Agent.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+Moreover, thanks to the notification system, one gets quickly informed about the main events, such as
+peering's status or the arrival of new Advertisements.
+
+### Features
+*   **Status information**: get details about Liqo:
+    * Liqo turned on/off
+    * Liqo work mode 
+    * active peerings
+*   **Start/Stop** Liqo
+*   **Choose** Liqo **work mode**:
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:24]: The Working mode are the key abstractions that the agent build on Liqo, in particular for Desktop users.

File: docs/design/Liqo-Agent.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+*   **Peerings** and **Advertisements management**: using a '_networks management -like_' interface,
+users can monitor available Liqo peers, read the received Advertisements (offers) 
+and manage the connections.
+
+### Limitations
+
+In order to orchestrate the tray icon and menu, _Agent_ exploits the [systray](https://github.com/getlantern/systray)
+package which has some limitations:
+* Missing sub-menu implementation. _Agent_ overcame the problem using native OS graphic windows.
+* The tray menu works as a stack, and the library offers no possibility to delete an Item.
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:51]: Those are limitations from an implementation point of view. I feel that this section should be more oriented about what user that install the agent today cannot perform (e.g.; start/stop liqo).

File: docs/design/Liqo-Agent.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+package which has some limitations:
+* Missing sub-menu implementation. _Agent_ overcame the problem using native OS graphic windows.
+* The tray menu works as a stack, and the library offers no possibility to delete an Item. 
+_Agent_ partially solved the problem changing properly the menu items visibility.
+
+## Architecture and workflow
+
+Take a look at the complete api [documentation](https://pkg.go.dev/github.com/liqoTech/liqo/internal/tray-agent).
+
+### Components
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:58]: I would rename/wrap this section on a implementation part.

File: docs/design/Liqo-Agent.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+* get **quick access** to Liqo **main features**, e.g. start/stop, Liqo working mode,
+peering management
+* manage the Agent settings, e.g. notifications
+
+_Agent_ aims at enhancing the Liqo users' experience, especially for non-pro ones, 
+allowing them to perform simple operations without needing to interact with the **LiqoDash**.
+Moreover, thanks to the notification system, one gets quickly informed about the main events, such as
+peering's status or the arrival of new Advertisements.
+
+### Features
```
Comment (palexster) [inline docs/design/Liqo-Agent.md:18]: Are those features already available?

File: docs/design/Liqo-Agent.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (AbakusW) [reply docs/design/Liqo-Agent.md:24]: Hi fraborg! After the last meeting, there was a proposal to put the Slave ("Dumb") mode implementation on hold.

**Pull request merged**